spec/helpers.lua

```diff
@@ -49,7 +110,19 @@ function resty_http_proxy_mt:send(opts)
     opts.query = nil
   end
 
-  return self:request(opts)
+  local res, err = self:request(opts)
+  if res then
+    -- wrap the read_body() so it caches the result and can be called multiple times
+    local reader = res.read_body
+    res.read_body = function(self)
+      if (not self._cached_body) and (not self._cached_error) then
```

Comment:
due to operator precedence, there can be very subtle bugs in a comparison involving `and`, `or` and `not`. Hence its good practice to use parenthesis to ensure it works according to the intentions and make it easier to read the code.